ensight_lib/include/bbox.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2016 Fraunhofer ITWM
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+#ifndef BBOX_H
+#define BBOX_H
+
+#include "eigentypes.h"
+
+/**
+ * @brief Represents a bounding box.
+ *
+ * This class is used to represent geometric bounds which is required for the Octree.
+ *
+ * This class is also used to represent bounds of variables.
+ * For scalar variables the x-component of min_ and max_ are used and the y-, and z-component are ignored.
+ */
+class Bbox
```

Comment:
It does most of what we need, although not with quite the same interface (e.g. it has no center() method).
It would probably make sense to have this class wrap Eigen::AlignedBox instead of 2 Vector3d, though. I'm not sure what the original reason was not to use Eigen::AlignedBox class, but for backwards compatibility (use of this library in existing internal code) I would prefer not to change the interface for now.